lib/types.go

```diff
@@ -155,38 +156,46 @@ func (s *containerCertificationSubmitter) Submit(ctx context.Context) error {
 		// The certification engine writes the rpmManifest for images not based on scratch.
 		WithRPMManifest(rpmManifest).
 		// Include the preflight execution log file.
-		WithArtifact(logfile, filepath.Base(s.preflightLogFile))
+		WithArtifact(logfile, filepath.Base(s.PreflightLogFile))
 
 	input, err := submission.Finalize()
 	if err != nil {
 		return fmt.Errorf("unable to finalize data that would be sent to pyxis: %w", err)
 	}
 
-	certResults, err := s.pyxis.SubmitResults(ctx, input)
+	certResults, err := s.Pyxis.SubmitResults(ctx, input)
 	if err != nil {
 		return fmt.Errorf("could not submit to pyxis: %w", err)
 	}
 
 	log.Info("Test results have been submitted to Red Hat.")
 	log.Info("These results will be reviewed by Red Hat for final certification.")
 	log.Infof("The container's image id is: %s.", certResults.CertImage.ID)
-	log.Infof("Please check %s to view scan results.", buildScanResultsURL(s.certificationProjectID, certResults.CertImage.ID))
-	log.Infof("Please check %s to monitor the progress.", buildOverviewURL(s.certificationProjectID))
+	log.Infof("Please check %s to view scan results.", BuildScanResultsURL(s.CertificationProjectID, certResults.CertImage.ID))
+	log.Infof("Please check %s to monitor the progress.", BuildOverviewURL(s.CertificationProjectID))
 
 	return nil
 }
 
 // noopSubmitter is a no-op resultSubmitter that optionally logs a message
 // and a reason as to why results were not submitted.
-type noopSubmitter struct {
+type NoopSubmitter struct {
 	emitLog bool
 	reason  string
 	log     *log.Logger
 }
 
-var _ resultSubmitter = &noopSubmitter{}
+func NewNoopSubmitter(emitLog bool, reason string, log *log.Logger) *NoopSubmitter {
+	return &NoopSubmitter{
+		emitLog: emitLog,
+		reason:  reason,
+		log:     log,
+	}
+}
```

Comment:
If we're going to export the struct, we may as well export the fields and remove the setters, I think?

Reply:
That makes logical sense to me as well.

Reply:
I think this also could be kept internal now using @komish's proposal.  👍